packages/next/server/hot-reloader.js

```diff
@@ -164,10 +164,18 @@ export default class HotReloader {
     return del(join(this.dir, this.config.distDir), { force: true })
   }
 
-  addWsPort (configs) {
-    configs[0].plugins.push(new webpack.DefinePlugin({
+  addWsConfig (configs) {
+    const opts = {
       'process.env.NEXT_WS_PORT': this.wsPort
-    }))
+    }
+    const { websocketProxyPath, websocketProxyPort } = this.config.onDemandEntries
+    if (websocketProxyPath) {
+      opts['process.env.NEXT_WS_PROXY_PATH'] = websocketProxyPath
```

Comment:
@timneutkens Do you think I should change this line too now that we've updated `on-demand-entries-client` - i.e. would it be better to always define `process.env.NEXT_WS_PROXY_PATH` and `process.env.NEXT_WS_PROXY_PORT` based on their default or user defined values? As it stands right now, I think `wsPath` would add `undefined` to the end of the socket path.

Reply:
I was thinking the same thing, I just pushed up a commit to both fix the setting / make test pass + change this!